src/entity/Entity.js

```diff
@@ -17,6 +17,31 @@ const variableToObject = (obj) => {
   return obj;
 };
 
+/**
+ * This function checks if the type of param is among the list : expectedTypes
+ * @param {*} expectedTypes the list with the expected types
+ * @param {*} param the parameter which has to be checked
+ * @returns a boolean - true if typeof param is among the list, false if not
+ */
+const checkType = (expectedTypes, param) => {
+  let paramHasAnExpectedType = false;
+  expectedTypes.forEach(
+    (type) => {
+    // we check if the type is a primitive type : ['string','number','boolean',...]
+      if (typeof (type) === 'string') {
+        if (expectedTypes.includes(typeof (param))) {
+        // we check if the param has a right expected primitive Type
+          paramHasAnExpectedType = true;
+        }
+      } else if ((param instanceof type)) { // we check if the param has a right expected Type
+      // the type is not a primitive type
+        paramHasAnExpectedType = true;
+      }
+    },
+  );
+  return paramHasAnExpectedType;
```

Comment:
why don't you throw there? I would rename the function to throwIfTheParameterHasNotTheExpectedType